test/utilities/testsUtils.ts

```diff
@@ -0,0 +1,224 @@
+  // Verify the tests that are passing are labeled with a green dot on the Test sidebar
+  const icon = await (await testItem.elem).$('.custom-view-tree-node-item-icon');
+  const iconStyle = await icon.getAttribute('style');
+  // Try/catch used to get around arbitrary flaky failure on Ubuntu in remote
+  try {
+    await expect(iconStyle).toContain(colorLabel);
+  } catch {
+    log('ERROR: icon did not turn green after test successfully ran');
+  }
+}
+
+/**
+ * Verifies that the test result contains all expected text snippets.
+ *
+ * @param {string} testResult - The actual test result as a string that needs to be verified.
+ * @param {string[]} expectedTexts - An array of strings representing the expected text snippets that should be present in the test result.
+ *
+ * @example
+ * await verifyTestResult(
+ *   testResult,
+ *   [
+ *     '=== Test Summary',
+ *     'Outcome              Passed',
+ *     'Tests Ran            1',
+ *     'Pass Rate            100%',
+ *     'TEST NAME',
+ *     'ExampleTest1  Pass',
+ *     'ended SFDX: Run Apex Tests'
+ *   ]
+ * );
+ */
+export async function verifyTestResult(testResult: string, expectedTexts: string[]): Promise<void> {
+  for (const expectedText of expectedTexts) {
+    await expect(testResult).toContain(expectedText);
+  }
+}
+
+/**
+ * Verifies the presence of test items in the sidebar.
+ * *
+ * @param {ViewSection} testsSection - An instance of the view section representing the sidebar where test items are displayed.
+ * @param {string} refreshCommand - The command used to refresh the sidebar to ensure it displays up-to-date information.
+ * @param {string[]} expectedItems - An array of strings representing the expected test items that should be present in the sidebar.
+ * @param {number} expectedTests - The expected number of tests to be displayed in the sidebar.
+ * @param {number} expectedClasses - The expected number of test classes to be present in the sidebar.
+ *
+ * @returns {Promise<void>} A promise that resolves when the verification is complete.
+ *
+ * @example
+ * await verifyTestItemsInSideBar(
+ *   mySidebarSection,
+ *   'Refresh Tests',
+ *   ['Test Item 1', 'Test Item 2'],
+ *   2,
+ *   1
+ * );
+ */
+export async function verifyTestItemsInSideBar(
+  testsSection: ViewSection,
+  refreshCommand: string,
+  expectedItems: string[],
+  expectedTests: number,
+  expectedClasses: number
+) {
+  log('Starting verifyTestItemsInSideBar()');
+  const testsItems = await retrieveExpectedNumTestsFromSidebar(
+    expectedTests,
+    testsSection,
+    refreshCommand
+  );
+  const isLWCSection = refreshCommand.includes('Lightning');
+  if (isLWCSection) {
+    log('Running LWC Tests');
+    // Expand tests
+    for (let x = 0; x < expectedClasses; x++) {
+      await testsItems[x].expand();
+    }
+  }
+
+  // Make sure all the tests are present in the sidebar
+  await expect(testsItems.length).toBe(isLWCSection ? expectedClasses : expectedTests);
```

Comment:
Given that the LWC Tests section is expanded in Lines 199-205, this doesn't make sense to me.  Why aren't we checking for `expectedTests` for both the Apex Tests and LWC Tests sections?

Reply:
They behave differently🤷🏻‍♀️ When you refresh apex tests, they show up expanded. When you refresh LWC tests, they show up collapsd. It's the same expectation than before the refactor, I didn't change that. That will have to change with [W-16428742](https://gus.lightning.force.com/a07EE00001yKRUPYA4).

Reply:
Ah I see.  `testsItems` is defined as the number of entries in the sidebar, which is only the LWC classes because the section is collapsed.  Lines 199-205 to expand each section comes _after_ `testsItems` is defined, and `testsItems` is not modified to include the new entries after expansion. ✅